controllers/v2/unavailability/api.py

```diff
@@ -1,31 +1,33 @@
-
 from flask import Blueprint, request, jsonify
 from flask_restful import reqparse, Resource, fields, marshal_with, Api, inputs
 
-from domain import session_scope
+from domain import session_scope, UserType
 from repository.volunteer_unavailability_v2 import *
-
+from services.jwk import requires_auth, is_user_or_has_role
+from controllers.v2.v2_blueprint import v2_api
 
 edit_parser = reqparse.RequestParser()
-edit_parser.add_argument("userId", type=int, required=True)
-edit_parser.add_argument("eventId", type=int, required=True)
 edit_parser.add_argument("title", type=str)
 edit_parser.add_argument("start", type=inputs.datetime_from_iso8601)
 edit_parser.add_argument("end", type=inputs.datetime_from_iso8601)
 edit_parser.add_argument("periodicity", type=int)
 
-class volunteer_unavailability(Resource):
-    def patch(self):
+
+class VolunteerUnavailabilityV2(Resource):
+
+    @requires_auth
+    @is_user_or_has_role(None, UserType.ROOT_ADMIN)
```

Comment:
This checks if the user is either (a) editing their own data, or (b) has the role required to edit someone elses